test/unit/annotation_spec.js

```diff
@@ -2317,6 +2317,28 @@ describe("annotation", function () {
           done();
         }, done.fail);
     });
+
+    it("should save nothing", function (done) {
+      const buttonWidgetRef = Ref.get(124, 0);
+      const xref = new XRefMock([
+        { ref: buttonWidgetRef, data: buttonWidgetDict },
+      ]);
+      const task = new WorkerTask("test save");
+
+      AnnotationFactory.create(
+        xref,
+        buttonWidgetRef,
+        pdfManagerMock,
+        idFactoryMock
+      )
+        .then(annotation => {
+          return annotation.save(partialEvaluator, task, {});
+        }, done.fail)
+        .then(data => {
+          expect(data).toEqual(null);
+          done();
+        }, done.fail);
```

Comment:
Nit: Do you actually need to repeat the `done.fail` part multiple times here?
Unless I'm missing something obvious here, it ought to be possible to simply add one `catch` at the end to deal with failures:
```suggestion
        .then(annotation => {
          return annotation.save(partialEvaluator, task, {});
        })
        .then(data => {
          expect(data).toEqual(null);
          done();
        })
        .catch(done.fail);
```